src/SourceBuild/Arcade/eng/common/templates/job/source-build-build-tarball.yml

```diff
@@ -55,15 +63,30 @@ jobs:
     - task: Bash@3
       displayName: Setup Private Feeds Credentials
       inputs:
-        filePath: $(Build.SourcesDirectory)/eng/common/SetupNugetSources.sh
-        arguments: $(Build.SourcesDirectory)/NuGet.config $Token
+        filePath: $(installerSrcDir)/eng/common/SetupNugetSources.sh
```

Comment:
I think the problem with the failing tests is that in an offline build, we strip the online feeds from the repo's nuget configs - https://github.com/dotnet/installer/blob/main/src/SourceBuild/tarball/content/repos/Directory.Build.targets#L184.  Additionally, we don't want to be adding these sources for the installer's nuget.config we build with.  We really only want to manipulate the nuget.config used for testing.  I'll let you propose how to accomplish that.